src/database/repositories/CacheRepository.ts

```diff
@@ -0,0 +1,111 @@
+import { IGuildSchema } from '@utils/Types';
+import { Redis } from 'ioredis';
+import { Document } from 'mongoose';
+import GuildsRepository from './GuildsRepository';
+
+export default class CacheRepository {
+  constructor(private redisClient: Redis | null, private guildRepository: GuildsRepository) {}
+
+  async fetchGuild(guildID: string): Promise<IGuildSchema | (IGuildSchema & Document)> {
+    if (this.redisClient) {
+      const guildData = await this.redisClient.get(`guild:${guildID}`);
+      if (guildData) return JSON.parse(guildData);
+    }
+    const guildDataFromMongo = await this.guildRepository.findOrCreate(guildID);
+
+    if (this.redisClient)
+      await this.redisClient.set(
+        `guild:${guildID}`,
+        JSON.stringify({
+          prefix: guildDataFromMongo.prefix,
+          lang: guildDataFromMongo.lang,
+          blockedChannels: guildDataFromMongo.blockedChannels,
+          disabledCommands: guildDataFromMongo.disabledCommands,
+        }),
+      );
+
+    return guildDataFromMongo;
+  }
+
+  async enableCommandInGuild(guildID: string, commandName: string): Promise<void> {
+    if (this.redisClient) {
+      const dataAtual = await this.redisClient.get(`guild:${guildID}`);
+      if (dataAtual) {
+        const jsonData = JSON.parse(dataAtual);
+        const index = jsonData.disabledCommands.indexOf(commandName);
+        jsonData.disabledCommands.splice(index, 1);
+        const afterChanges = JSON.stringify(jsonData);
+        await this.redisClient.set(`guild:${guildID}`, afterChanges);
+      }
+    }
```

Comment:
IMO this pattern is cleaner, and it uses one less scope. If you like it as well, you can replicate it on the rest of the methods in this file

```suggestion
    redisBlock: if (this.redisClient) {
      const cachedData = await this.redisClient.get(`guild:${guildID}`);
      if (!cachedData) break redisBlock;
      const jsonData = JSON.parse(cachedData);
      const index = jsonData.disabledCommands.indexOf(commandName);
      jsonData.disabledCommands.splice(index, 1);
      await this.redisClient.set(`guild:${guildID}`, JSON.stringify(jsonData));
    }
```

Reply:
I tried to use it, but it breakes 2 rules of my eslint, so i wont use it, but thanks for the suggestion, i didnt know the tabled statements, it seems to be very util